ddtrace/contrib/django/patch.py

```diff
@@ -666,14 +674,188 @@ def django_asgi_modifier(span, scope):
     return TraceMiddleware(func(*args, **kwargs), integration_config=config.django, span_modifier=django_asgi_modifier)
 
 
+_POSSIBLE_USER_ID_FIELDS = ["pk", "id", "uid", "userid", "user_id", "PK", "ID", "UID", "USERID"]
+_POSSIBLE_LOGIN_FIELDS = ["username", "user", "login", "USERNAME", "USER", "LOGIN"]
+_POSSIBLE_EMAIL_FIELDS = ["email", "mail", "address", "EMAIL", "MAIL", "ADDRESS"]
+_POSSIBLE_NAME_FIELDS = ["name", "fullname", "full_name", "NAME", "FULLNAME", "FULL_NAME"]
+
+
+def _find_in_user_model(user, possible_fields):
+    for field in possible_fields:
+        value = getattr(user, field, None)
+        if value:
+            return value
+
+    return None  # explicit to make clear it has a meaning
+
+
+def _get_userid(user):
+    user_login = getattr(user, config._user_model_login_field, None)
+    if user_login:
+        return user_login
+
+    return _find_in_user_model(user, _POSSIBLE_USER_ID_FIELDS)
+
+
+def _get_username(user):
+    username = getattr(user, config._user_model_name_field, None)
+    if username:
+        return username
+
+    if hasattr(user, "get_username"):
+        return user.get_username()
+
+    user_type = type(user)
+    if hasattr(user_type, "USERNAME_FIELD"):
+        return getattr(user, user_type.USERNAME_FIELD, None)
+
+    return _find_in_user_model(user, _POSSIBLE_LOGIN_FIELDS)
+
+
+def _get_user_email(user):
+    email = getattr(user, config._user_model_email_field, None)
+    if email:
+        return email
+
+    user_type = type(user)
+    if hasattr(user_type, "EMAIL_FIELD"):
+        return getattr(user, user_type.EMAIL_FIELD, None)
+
+    return _find_in_user_model(user, _POSSIBLE_EMAIL_FIELDS)
+
+
+def _get_name(user):
+    if hasattr(user, "get_full_name"):
+        return user.get_full_name()
+
+    if hasattr(user, "first_name") and hasattr(user, "last_name"):
+        return "%s %s" % (user.first_name, user.last_name)
+
+    return getattr(user, "name", None)
+
+
+def _get_user_info(user):
+    """
+    In safe mode, try to get the user id from the user object.
+    In extended mode, try to also get the username (which will be the returned user_id),
+    email and name.
+
+    Since the Django Authentication model supports pluggable users models
+    there is some heuristic involved in extracted this field, through it should
+    work well for the most common case with the default User model.
+    """
+    safe_mode = config._automatic_login_events_mode == "safe"
+    user_extra_info = {}
+
+    if safe_mode:
+        user_id = _get_userid(user)
+    else:
```

Comment:
So anything other than `safe`, e.g. `extended` and we'll get extended behavior, correct?

Reply:
That is actually a nasty bug you got. I will fix this and also add a test checking that safe is always the default.